lib/nats.js

```diff
@@ -1534,29 +1545,33 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
         msg = msg === undefined ? null : msg;
     }
 
+    if (opt_callback && typeof opt_callback !== 'function') {
+        throw (new NatsError(BAD_CALLBACK_MSG, BAD_CALLBACK));
+    }
+
     if (!subject) {
         if (opt_callback) {
             opt_callback(new NatsError(BAD_SUBJECT_MSG, BAD_SUBJECT));
+            return;
```

Comment:
This missing return lead to `opt_callback` being called twice, once here with an error and one more time later, without an error.